src/rgw/rgw_lc.cc

```diff
@@ -1457,15 +1418,17 @@ class LCOpAction_Transition : public LCOpAction {
     if (ret < 0) {
       return ret;
     } else {
-      // send request to notification manager
-      int publish_ret =  notify->publish_commit(oc.dpp, obj_state->size,
-				    ceph::real_clock::now(),
-				    etag,
-				    version_id);
-      if (publish_ret < 0) {
-	ldpp_dout(oc.dpp, 5) <<
-	  "WARNING: notify publish_commit failed, with error: " << publish_ret << dendl;
+      rgw::notify::EventTypeList event_types;
+      if (bucket->versioned() && oc.o.is_current()
```

Comment:
Unrelated to this PR , there is a bug here -
Like in delete_tier_obj()  routine above (https://github.com/ceph/ceph/pull/57079/files#diff-4e9c90acd6086f07576e1849e3785731dc77b0e5c42bc76b16b4a263e8090ef7L1379), there should have been one more case here to check if the bucket is not versioned in which case notify should be of type `rgw::notify::ObjectTransition`.

With the current code, `rgw::notify::ObjectTransitionNoncurrent` will be sent for all the non-versioned objects too.

Reply:
@soumyakoduri not sure if understand your comment, looking the [file](https://github.com/ceph/ceph/blob/745beaed88251a86b1ab322c938c80b245d0ed4c/src/rgw/rgw_lc.cc) that is part of this PR,
i see this [line](https://github.com/ceph/ceph/blob/745beaed88251a86b1ab322c938c80b245d0ed4c/src/rgw/rgw_lc.cc#L1366) covers --> rgw::notify::ObjectTransition (for non versioned bucket)
this [line](https://github.com/ceph/ceph/blob/745beaed88251a86b1ab322c938c80b245d0ed4c/src/rgw/rgw_lc.cc#L1373) covers --> rgw::notify::ObjectTransitionCurrent
and this [line](https://github.com/ceph/ceph/blob/745beaed88251a86b1ab322c938c80b245d0ed4c/src/rgw/rgw_lc.cc#L1381) covers non-current --> rgw::notify::ObjectTransitionNoncurrent

so which use case is missing   ?

Reply:
I mean like you mentioned above, it is handled in `delete_tier_obj()` (the routine which removes the cloud transitioned object) but not in this generic `transition` code ([line](https://github.com/ceph/ceph/blob/745beaed88251a86b1ab322c938c80b245d0ed4c/src/rgw/rgw_lc.cc#L1422)). Here if the bucket is not versioned, the notification event generated will be `rgw::notify::ObjectTransitionNoncurrent`. 

Anyway hopefully this will get addressed as part of https://github.com/ceph/ceph/pull/57551#discussion_r1606805069

Reply:
@soumyakoduri thanks.
Done, added new commit

Reply:
@kchheda3 ..   [PR#57551](https://github.com/ceph/ceph/pull/57551/files#diff-4e9c90acd6086f07576e1849e3785731dc77b0e5c42bc76b16b4a263e8090ef7R1476) addresses this issue as well. You may omit the new commit. Rest all changes look good to me.

Reply:
> @kchheda3 .. [PR#57551](https://github.com/ceph/ceph/pull/57551/files#diff-4e9c90acd6086f07576e1849e3785731dc77b0e5c42bc76b16b4a263e8090ef7R1476) addresses this issue as well. You may omit the new commit. Rest all changes look good to me.

done

Reply:
> [PR#57551](https://github.com/ceph/ceph/pull/57551/files#diff-4e9c90acd6086f07576e1849e3785731dc77b0e5c42bc76b16b4a263e8090ef7R1476)

can we merge this first to avoid the rebase coz of [PR#57551](https://github.com/ceph/ceph/pull/57551/files#diff-4e9c90acd6086f07576e1849e3785731dc77b0e5c42bc76b16b4a263e8090ef7R1476)